spec/controllers/settings/ldap_test_controller_spec.rb

```diff
@@ -0,0 +1,136 @@
+require "rails_helper"
+require "json"
+
+RSpec.describe Settings::LdapTestController, type: :controller do
+    let(:user) { create(:user) }
+  
+    before do
+      setup_done
+      sign_in user
+    end
+
+    describe "POST #create" do
+        let(:host) { "ldaptest.com" }
+        let(:port) { 636 }
+        let(:start_tls) { "false" }
+        let(:cert) { "MIIC0zCCAlmgAwIBAgIUCfQ+m0pgZ/BjYAJvxrn/bdGNZokwCgYIKoZIzj0EAwMw
```

Comment:
It may make sense to use TLS library directly to fetch the cert from the host at the very start, so that whatever cert ldaptest.com provides it still functions.

Reply:
Added code to the RSpec test (ldap_test_controller_spec.rb) which gets the necessary cert from the LDAP server.